cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -378,6 +379,9 @@ type ControllerContext struct {
 
 	// ControllerManagerMetrics provides a proxy to set controller manager specific metrics.
 	ControllerManagerMetrics *controllersmetrics.ControllerManagerMetrics
+
+	// GraphBuilder gives an access to dependencyGraphBuilder which keeps tracks of resources in the cluster
+	GraphBuilder *garbagecollector.GraphBuilder
```

Comment:
You need an interfaces that does these things, right?

WaitForMonitorSync
IsSynced
GetMonitorController
GetMonitorStore

Reply:
based on live conversation, this is ok for alpha.  But we need the garbage collection controller to manage it still for alpha.

Prior to beta, we must end up in a state where this is separate and started if either garbagecollector or storageversionmigrator is active